Cargo.toml

```diff
@@ -31,6 +31,9 @@ features = ["span-locations"]
 [dependencies]
 unicode-ident = "1.0"
 
+[target.'cfg(procmacro2_semver_exempt)'.dependencies]
+rustc-literal-escaper = "0.0.5"
```

Comment:
This dependency's compiler version support is not compatible with making it a dependency of proc-macro2. Also it has unreasonably many publishers for something that would become a widely used dependency. Neither of these factors is necessarily immediately disqualifying while the dependency is gated by our semver exempt cfg, but they mean that this implementation is not going to work once it comes time to stabilize the new methods. I would prefer going directly to a different implementation that does not involve adding this dependency.